draft-ietf-quic-transport.md

```diff
@@ -494,15 +488,15 @@ other fields.
 
 | Type | Packet Number Size |
 |:-----|:-------------------|
-| 0x1F | 1 octet            |
-| 0x1E | 2 octets           |
-| 0x1D | 4 octets           |
+| 0x3F | 1 octet            |
+| 0x3E | 2 octets           |
+| 0x3D | 4 octets           |
 {: #short-packet-types title="Short Header Packet Types"}
 
-The header form, omit connection ID flag, and connection ID of a short header
-packet are version-independent.  The remaining fields are specific to the
-selected QUIC version.  See {{version-specific}} for details on how packets
-from different versions of QUIC are interpreted.
+The header form and connection ID of a short header packet are
+version-independent.  The remaining fields are specific to the selected QUIC
```

Comment:
This makes the Connection ID in the short header a difficult invariant.  There might be one, but there's no way to identify how long it is.  Middleboxes that want to use the CID for routing, unless they're in active collaboration with the server, are kind of stuck.  Which might be intentional, but it seems awkward.

Reply:
I think that the purpose of the connection ID is for cooperating intermediaries, not random middleboxen.  I'm sure that others will disagree, as we're seeing with the proposal to encrypt packet numbers.

Reply:
That change is integral to the PR.  The bit only makes sense if you intend to send the connection ID on some packets but not others and if you think that an intermediary that is ignorant of the actual length being able to do something meaningful with both sets of packets.

Reply:
That's not the scenario I outlined in my comment. You run a server farm and purchase a middlebox from a vendor. If you have a connection ID bit, you don't have to configure the middlebox to align with server policy about connection ID, because packets are self-describing.

The bit makes sense only if the size is known to both the server and the middlebox however, so you have a good point. Both entities have to agree on a way to determine the size of the connection ID, and as we've discussed earlier, that can be achieved by making the connection ID a varlen. The spec doesn't have to make it a varlen. A middlebox can simply require that connection IDs are vended with a particular encoding by servers, and that's adequate.

I don't want to design this here; I'm noting that self-describing packets have value. I think this should be discussed separately from this PR. That said